lib/fs.js

```diff
@@ -1469,9 +1471,15 @@ function readdirSyncRecursive(basePath, options) {
       }
     }
   }
-
+  const base = basePath.replace(/\//g, sep) + sep;
   for (let i = 0; i < pathsQueue.length; i++) {
-    read(pathsQueue[i]);
+    if (
+      options.depth < 1 ||
+      pathsQueue[i] === basePath ||
+      pathsQueue[i].replace(base, '').split(sep).length < options.depth
```

Comment:
This likely has a not so great performance, couldn't we keep track of how deep we are instead using a variable that we increment?

Reply:
The current implementation makes this confusing.  The async function fs.readdir currently uses the synchronous `readdirSyncRecursive` to apply recursion.  In my own personal library where I worked this out in the past its always asynchronous because I perform a different fs.readdir on each child directory, so I expect performance to differ there versus here.  In my own code though, this approach proved insignificantly slow because its always less slow than transitions through the event loop on each readdir call.

In order to keep track of depth you would have to keep track per each descendant directory.  That is why implemented depth checks in the first place, because such a tracking tree can become exceedingly large.